src/geo/transform.js

```diff
@@ -239,12 +239,36 @@ class Transform {
     }
 
     /**
-     * Given a point on screen, return its lnglat
+     * Given a point on screen and viewport options, return its lnglat
      * @param {Point} p screen point
+     * @param {CameraOptions} v viewport
      * @returns {LngLat} lnglat location
      */
-    pointLocation(p) {
-        return this.coordinateLocation(this.pointCoordinate(p));
+    pointLocation(p, v) {
+        if (v == null) {
+            return this.coordinateLocation(this.pointCoordinate(p));
+        } else {
+            const originalViewport = {
+                center: this.center,
+                zoom: this.zoom,
+                bearing: this.bearing,
+                pitch: this.pitch
+            };
+
+            if ('center' in v) this.center = v.center;
+            if ('zoom' in v) this.zoom = v.zoom;
+            if ('bearing' in v) this.bearing = v.bearing;
+            if ('pitch' in v) this.pitch = v.pitch;
```

Comment:
I get what you mean by that, I looked into it. I could add an extra argument to the `coordinateLocation` and `pointCoordinate` methods but the reason why I used this implicit setting of parameters is to make sure that the pixelMatrix is calculated for the viewport. What do you suggest that I should do concerning the `_calcMatrices` function, should I rewrite it to allow a viewport parameter? `_calcMatrices` uses the following fields:

- `this.x`
- `this.y`
- `this._fov`
- `this._pitch`
- `this.width`
- `this.height`
- `this.worldSize`

If I'm going to do it with a viewport parameter I'm also a bit afraid to introduce a lot of duplicate code because every parameter has a seperate check and a lot of other `Transform` fields are dependent on the ones used in the viewport such as `this.worldSize` which is dependent on `this.scale` which is dependent on `this.zoom`.